### DISCUSSION TITLE
Is it possible to checkout an arbitrary commit?

### DISCUSSION THREAD

**Discussion by Miyelsh** [NeogitOrg/neogit#1438] [Q&A]
I can use `<bb>` to checkout named branches, but what if I want to go to an arbitrary commit, or even something simple like `HEAD^`.

Similarly, if I want to get the log of a specific directory, I can't find a way to do it. I can select a specific file, but what about a specific directory?

These things I end up using the command line to do, is there a way to do it within neogit?

**Comment by CKolkey** [comment]
With your cursor over /any/ commit-ish item, in any neogit view, `bb` (and many others) will take that as input and it will be available as the first item. So, to do a specific commit, you can find it in the log, or if its recent then it is on the status buffer.

Similarly, from the status buffer, the HEAD lines up top work, and the current HEADs are in the list
